src/filefinder.cpp

```diff
@@ -646,26 +646,37 @@ void FileFinder::Quit() {
 	game_directory_tree.reset();
 }
 
-FILE* FileFinder::fopenUTF8(const std::string& name_utf8, char const* mode) {
-#ifdef _WIN32
-	return _wfopen(Utils::ToWideString(name_utf8).c_str(),
-				   Utils::ToWideString(mode).c_str());
+Filesystem::InputStream FileFinder::OpenInputStream(const std::string& name,
+	std::ios_base::openmode m)
+{
+	std::streamsize size = FileFinder::GetFileSize(name);
```

Comment:
You should be able to get the file size from the `filebuf`, this can save you from having to open the file twice.

Reply:
I'm a bit indifferent here. Which of the ways if faster should be also explored via benchmarking. 
At least for compressed streams seek-to-end and back to begin is bad because it forces a full decompression that could be avoided in some cases.

Also the file size isn't really used a lot except for some buffer preallocation optimisations. Imo we can remove it completely later and just read until eof (but needs more testing if this breaks some libs)